drivers/regulator/regulator_axp192.c

```diff
@@ -0,0 +1,408 @@
+/*
+ * Copyright (c) 2023 M5Stack
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT xpowers_axp192_regulator
+
+#include <errno.h>
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/linear_range.h>
+#include <zephyr/sys/util.h>
+
+/* X-Powers AXP192 chip ID */
+#define AXP192_CHIP_ID 0x03
+
+/* X-Powers AXP192 register */
+#define AXP192_REG_POWER_STATUS      0x00
+#define AXP192_REG_CHIP_ID           0x03
+#define AXP192_REG_BUCK123_LDO23_CTL 0x12
+#define AXP192_REG_BUCK1_OUT_VOL     0x26
+#define AXP192_REG_BUCK2_OUT_VOL     0x23
+#define AXP192_REG_BUCK3_OUT_VOL     0x27
+#define AXP192_REG_LDO23_OUT_VOL     0x28
+#define AXP192_REG_VBUS_IPS_CTL      0x30
+#define AXP192_REG_CHG_CTL           0x33
+#define AXP192_REG_PER_KEY_CTL       0x36
+#define AXP192_REG_ACIN_IN_VOL       0x56
+#define AXP192_REG_ACIN_IN_CUR       0x58
+#define AXP192_REG_VBUS_IN_VOL       0x5A
+#define AXP192_REG_VBUS_IN_CUR       0x5C
+#define AXP192_REG_INTER_TEMP        0x5E
+#define AXP192_REG_BAT_VOL           0x78
+#define AXP192_REG_BAT_CHG_CUR       0x7A
+#define AXP192_REG_BAT_DCHG_CUR      0x7C
+
+/* X-Powers AXP192 voltage sources */
+enum axp192_voltage_sources {
+	AXP192_SOURCE_BUCK1,
+	AXP192_SOURCE_BUCK2,
+	AXP192_SOURCE_BUCK3,
+	AXP192_SOURCE_LDO1,
+	AXP192_SOURCE_LDO2,
+	AXP192_SOURCE_LDO3
```

Comment:
note: datasheet seems to advertise 4 LDOs:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/25011557/090c11f4-cba8-4242-829a-e654f1b3c198)

Reply:
Added